source/Missions/Services/Agents/Messages/ShieldHealthRemaining.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Common.Messaging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Messages
+{
+    /// <summary>
+    /// Internal event for agent shield damage
+    /// </summary>
+    public readonly struct ShieldHealthRemaining : IEvent
```

Comment:
Rename to `ShieldDamaged`
```suggestion
    public readonly struct ShieldDamaged : IEvent
```